.circleci/config.yml

```diff
@@ -264,6 +264,53 @@ jobs:
               make localnet-start
               ./scripts/localnet-blocks-test.sh 40 5 10 localhost
 
+  snap-edge:
+      working_directory: /home/circleci/.snap_workspace/src/github.com/cosmos/cosmos-sdk
+      machine: true
+      environment:
+        GOBIN: /home/circleci/.snap_workspace/bin
+        GOPATH: /home/circleci/.snap_workspace/
+        GOOS: linux
+        GOARCH: amd64
+        GO_VERSION: "1.11.4"
+        LC_ALL: C.UTF-8
+        LANG: C.UTF-8
+      parallelism: 1
+      steps:
+        - checkout
+        - run:
+            name: Install prerequitistes
+            command: |
+              sudo apt update -y &>/dev/null
+              sudo apt install -y build-essential openssl snapd &>/dev/null
+              sudo snap install snapcraft --edge --classic
+        - run:
+            name: prepping snap build
+            command: |
+              pushd /tmp
+              wget https://dl.google.com/go/go$GO_VERSION.linux-amd64.tar.gz
+              sudo tar -xf go$GO_VERSION.linux-amd64.tar.gz
+              sudo rm -rf /usr/local/go
+              sudo mv go /usr/local
+              popd
+              set -x
+              export PATH="$GOBIN:/usr/local/go/bin:$PATH"
+              /snap/bin/snapcraft version
+              make snapcraft.yaml
+        - run:
+            name: build edge snap
+            command: /snap/bin/snapcraft
+        - run:
+            name: Decrypt credentials
+            command: |
+              openssl aes-256-cbc -d -md sha256 -in .circleci/snapcraft_credentials.enc -out snapcraft_credentials -k $SNAPCRAFT_CREDENTIALS_KEY
```

Comment:
Can CircleCI store files like it does environment variables? If not, can we store this (snapcraft_credentials.enc) in ssm/kms? I think we should not store it in GitHub. Even, since it's in the PR, I think we should recreate it and store it somewhere else.

Reply:
CircleCI can store the credentials in environment variables if that's workable.

Or we can go the AWS SSM route. Either way, I'm also wary of storing credentials in github.